Python 3.13.0b3: AttributeError: 'functools.partial' object has no attribute '__name__'

```
$ gh repo clone encode/starlette
$ cd starlette
$ python3.13 --version
Python 3.13.0b3
$ python3.13 -m venv _e
$ . _e/bin/activate
(_e) $ pip install -r requirements.txt
(_e) $ pip install -e .
(_e) $ python -m pytest
_e/lib64/python3.13/site-packages/trio/_path.py:164: in generate_forwards
    raise TypeError(attr_name, type(attr))
E   TypeError: ('parser', <class 'module'>)
(_e) $ pip install --upgrade trio
(_e) $ python -m pytest
==================================================================================================== ERRORS ====================================================================================================
____________________________________________________________________________________ ERROR collecting tests/test_routing.py ____________________________________________________________________________________
tests/test_routing.py:135: in <module>
    Route("/", endpoint=functools.partial(partial_endpoint, "foo")),
starlette/routing.py:227: in __init__
    self.name = get_name(endpoint) if name is None else name
starlette/routing.py:103: in get_name
    return endpoint.__name__
E   AttributeError: 'functools.partial' object has no attribute '__name__'. Did you mean: '__ne__'?
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=============================================================================================== 1 error in 0.64s ===============================================================================================
```

This appears to be a new regression in Python 3.13.0b3 (vs. 3.13.0b2).

---

- [x] Initially raised as discussion #2610 – “Please file an issue in this repo for any incompatibility you encounter with 3.13.”

--- a/starlette/routing.py
+++ b/starlette/routing.py
@@ -99,9 +99,7 @@ def websocket_session(
 
 
 def get_name(endpoint: typing.Callable[..., typing.Any]) -> str:
-    if inspect.isroutine(endpoint) or inspect.isclass(endpoint):
-        return endpoint.__name__
-    return endpoint.__class__.__name__
+    return getattr(endpoint, "__name__", endpoint.__class__.__name__)
 
 
 def replace_params(

